src/Credentials/CredentialProvider.php

```diff
@@ -78,7 +79,10 @@ public static function defaultProvider(array $config = [])
         return self::memoize(
             self::chain(
                 self::env(),
+                self::assumeRole(),
```

Comment:
It's possible that people want to use the default credential chain, but also need to provide custom values to each chain depending on whether or not that chain is actually utilized... This might be a more complex problem that would need to be solved in a separate update, so maybe this can be left out for now.